OpenUpSA/wazimap-ng-ui
# error notifications

## Description
* catching all the exceptions in the app and showing an alert message to ask the user refresh the page.
* gui test to confirm an alert is shown or error

merged https://github.com/OpenUpSA/wazimap-ng-ui/pull/484 into this for the cypress related changes

## Related Issue
https://wazimap.atlassian.net/browse/WNCM-13


## How to test it locally
* create an exception in the page
    -> for example using the dev tools on the bottom left corner of the page, set the API url `http://www.test.com`
* confirm that an alert is shown asking to refresh the page


## Screenshots


## Changelog

### Added
* `error.js` to catch the exceptions and show an alert
* `__tests__/gui/error.feature` to test the new feature
* other changes are the result of merging https://github.com/OpenUpSA/wazimap-ng-ui/pull/484 into this

### Updated
* `index.js` to referance `error.js`. this must be done before initializing api
* `src/js/elements/subindicator_filter/dropdown.js` to console.error instead of throw
* `src/js/api.js` to console.error instead of throw

### Removed


## Checklist


### Pull Request


### Code Quality


### Testing
